### PR TITLE
RavenDB-11764 LoadStartingWith only returns first page when combined with matches parameter

### PR DESCRIPTION
**Pull request by haludi**
http://issues.hibernatingrhinos.com/issue/RavenDB-11764

`GetDocumentsWithIdStartingWith` modified to return `nextStart` also in the middle of the unfiltered page if there is the last document of filtered page and client modified to recognize the last page according to it

### CODE REVIEW

**Review by ayende — APPROVED** [review]

**Comment by arekpalinski** [comment]
Let's wait for CI test before we merge this one

**Comment by arekpalinski** [comment]
@haludi Please check failing test: 

Tests.MailingList.UsingLoadStartsWithAndDashSeperator.ShouldWork(storage: "esent")


MESSAGE:
Assert.Equal() Failure
Expected: 1
Actual:   2
+++++++++++++++++++
STACK TRACE:
   at Raven.Tests.MailingList.UsingLoadStartsWithAndDashSeperator.ShouldWork(String storage) in C:\Jenkins\workspace\PR_Tests_v3.5\s\Raven.Tests.MailingList\UsingLoadStartsWithAndDashSepartor.cs:line 34

**Pull request merged**
